mono/metadata/w32handle.c

```diff
@@ -199,7 +236,7 @@ mono_w32handle_unlock_signal_mutex (void)
 }
 
 void
-mono_w32handle_lock_handle (gpointer handle)
+mono_w32handle_lock_handle (gpointer handle, MonoW32HandleType expected_types)
```

Comment:
I wasn't planning on doing it for refs, as part of investigating the `pthread_cond_timedwait` crashes I a bad-type unref stacktrace is a bit more useful. This was when I briefly thought there was a path in this function where the `ref` had to be undone (like `mono_w32handle_trylock_handle`).